pkg/model/volumes.go

```diff
@@ -0,0 +1,217 @@
+	}
+	return dev.PersistentVolumeInfo.Enabled
+}
+
+// PersistentVolumeSize returns the persistent volume size
+func (dev *Dev) PersistentVolumeSize() string {
+	if dev.PersistentVolumeInfo == nil {
+		return OktetoDefaultPVSize
+	}
+	if dev.PersistentVolumeInfo.Size == "" {
+		return OktetoDefaultPVSize
+	}
+	return dev.PersistentVolumeInfo.Size
+}
+
+// PersistentVolumeStorageClass returns the persistent volume storage class
+func (dev *Dev) PersistentVolumeStorageClass() string {
+	if dev.PersistentVolumeInfo == nil {
+		return ""
+	}
+	return dev.PersistentVolumeInfo.StorageClass
+}
+
+func (dev *Dev) validatePersistentVolume() error {
+	if dev.PersistentVolumeEnabled() {
+		return nil
+	}
+	if len(dev.Services) > 0 {
+		return fmt.Errorf("'persistentVolume.enabled' must be set to true to work with services")
+	}
+	if dev.HasRemoteVolumes() {
+		return fmt.Errorf("'persistentVolume.enabled' must be set to true to use remote volumes")
+	}
+	for _, v := range dev.Volumes {
+		result, err := dev.IsSyncFolder(v.LocalPath)
+		if err != nil {
+			return err
+		}
+		if !result {
+			return fmt.Errorf("'persistentVolume.enabled' must be set to true to use subfolders in the 'volumes' field")
+		}
+	}
+	return nil
+}
+
+func (dev *Dev) validateVolumeRemotePaths() error {
+	for _, v := range dev.Volumes {
+		if !strings.HasPrefix(v.RemotePath, "/") {
+			return fmt.Errorf("relative remote paths are not supported as volumes")
+		}
+		if v.RemotePath == "/" {
+			return fmt.Errorf("remote path '/' is not supported as volumes")
+		}
+	}
+	return nil
+}
+
+func (dev *Dev) validateDuplicatedVolumes() error {
+	seen := map[string]bool{}
+	for _, v := range dev.Volumes {
+		key := v.LocalPath + ":" + v.RemotePath
+		if seen[key] {
+			if v.LocalPath == "" {
+				return fmt.Errorf("duplicated volume '%s'", v.RemotePath)
+			}
+			return fmt.Errorf("duplicated volume '%s:%s'", v.LocalPath, v.RemotePath)
+		}
+		seen[key] = true
+	}
+	return nil
+}
+
+func (dev *Dev) validateDuplicatedSyncFolders() error {
+	seen := map[string]bool{}
+	for _, v := range dev.Volumes {
+		if v.LocalPath == "" {
+			continue
+		}
+		result, err := dev.IsSyncFolder(v.LocalPath)
+		if err != nil {
+			return err
+		}
+		if !result {
+			continue
+		}
+		if seen[v.LocalPath] {
+			return fmt.Errorf("duplicated local volume '%s'", v.LocalPath)
+		}
+		seen[v.LocalPath] = true
+	}
+	return nil
+}
+
+func (dev *Dev) validateServiceSyncFolders(main *Dev) error {
+	for _, v := range dev.Volumes {
+		if v.LocalPath == "" {
+			continue
+		}
+		_, err := main.IsSyncFolder(v.LocalPath)
+		if err != nil {
+			return fmt.Errorf("Local volume '%s' in '%s' not defined in the main development container", v.LocalPath, dev.Name)
+		}
+	}
+	return nil
+}
+
+func (dev *Dev) validateVolumes(main *Dev) error {
+	if !dev.HasLocalVolumes() {
+		return fmt.Errorf("the 'volumes' field must define a local folder to be synched. More info at https://okteto.com/docs/reference/manifest#volumes-string-optional")
```

Comment:
on which condition would this error be triggered?

Reply:
when there is not `workdir` nor a Synched Volume define. It would mean nothing is synched. We can support this use case in the future